### PR TITLE
update python user guide on install and run yarn part

### LINKED ISSUE #1989: virtualenv: error: unrecognized arguments: --relocatable
https://github.com/intel-analytics/analytics-zoo/blob/master/scripts/python_package.sh#L29

Seems virtualenv has removed relocatable in recent versions:
https://virtualenv.pypa.io/en/16.7.9/userguide.html#making-environments-relocatable

Shall we remove it as well? Otherwise users would encounter this error. (I use 20.0.3)

created virtual environment in 312ms CPython3Posix(dest=/opt/work/client/kai/mastercard/analytics-zoo-bigdl_0.10.0-spark_2.4.3-0.7.0-dist-all/bin/venv, clear=False, global=False) with seeder FromAppData pip=latest setuptools=latest wheel=latest app_data_dir=/root/.local/share/virtualenv/seed-v1 via=copy
usage: virtualenv [--version] [--with-traceback] [-v | -q] [--discovery {builtin}] [-p py] [--creator {builtin,cpython3-posix,venv}] [--seeder {app-data,pip}] [--no-seed] [--activators comma_separated_list] [--clear]
                  [--system-site-packages] [--symlinks | --copies] [--download | --no-download] [--extra-search-dir d [d ...]] [--pip version] [--setuptools version] [--wheel version] [--no-pip] [--no-setuptools] [--no-wheel]
                  [--clear-app-data] [--symlink-app-data] [--prompt prompt] [-h]
                  dest
virtualenv: error: unrecognized arguments: --relocatable

### PR DESCRIPTION
**Pull request by jenniew**
Remove virtual environment doc for yarn cluster. Replace install and run pyspark in yarn cluster without pip document by using conda package. 

Fix #1989 as venv is fully removed.

### CODE REVIEW

File: docs/docs/PythonUserGuide/install.md [hunk 1]
```diff
@@ -100,32 +100,27 @@ Alternatively, you can also build the Analytics Zoo from [source](../ScalaUserGu
...
-and then directly use spark-submit to run the Analytics Zoo Python program on the Yarn cluster using that virtual environment.
+You can first package all the required dependencies into a conda environment on the local node (where you will run the spark-submit command),
+and then directly use spark-submit to run the Analytics Zoo Python program on the Yarn cluster using that conda environment.
 
-Follow the steps below to create the virtual environment: 
+Follow the steps below to create the conda environment:
    
-* Make sure you already installed such libraries (python-setuptools, python-dev, gcc, make, zip, pip) for creating the virtual environment. If not, please install them first.
-On Ubuntu, you can run these commands to install:
+1) Install [Conda](https://docs.conda.io/projects/conda/en/latest/commands/install.html) in your environment.
```
Comment (hkvision) [inline docs/docs/PythonUserGuide/install.md]: Can we provide an easy command to guide users how to install conda? Or provide a link?
This link is related to conda commands but not to install conda

File: docs/docs/PythonUserGuide/install.md [hunk 2]
```diff
@@ -100,32 +100,27 @@ Alternatively, you can also build the Analytics Zoo from [source](../ScalaUserGu
...
-apt-get install -y gcc make
-apt-get install -y zip
-easy_install pip
+conda create -n environment python=3.6 pip=20.0.2 setuptools=45.2.0 conda-pack==0.3.1
+conda activate environment
 ```
-* Create the virtualenv package for dependencies.
-    * ```export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package```
-    
-    * Run ```${ANALYTICS_ZOO_HOME}/bin/python_package.sh``` to create the dependency virtual environment according to the dependencies listed in `requirements.txt`. You can add your own dependencies into this file if you wish. The current requirements only contain those needed for running Analytics Zoo Python examples and models.
```
Comment (hkvision) [inline docs/docs/PythonUserGuide/install.md]: Keep or rephrase the sentence `You can add your own dependencies into this file if you wish. The current requirements only contain those needed for running Analytics Zoo Python examples and models.`?

File: docs/docs/PythonUserGuide/install.md [hunk 3]
```diff
@@ -100,32 +100,27 @@ Alternatively, you can also build the Analytics Zoo from [source](../ScalaUserGu
...
-In case you encounter the following errors when you create the environment package using the above command:
+4) Create conda package for the created conda environment
 
-1. virtualenv ImportError: No module named urllib3
-    - Using python in anaconda to create virtualenv may cause this problem. Try using python default in your system instead of installing virtualenv in anaconda.
-2. AttributeError: 'module' object has no attribute 'sslwrap'
-    - Try upgrading `gevent` with `pip install --upgrade gevent`.
+```
+conda pack -o environment.tar.gz
+```
```
Comment (hkvision) [inline docs/docs/PythonUserGuide/install.md]: At the end, add a sentence saying `you can find the file under the current directory` to guide users verify the package?

File: docs/docs/PythonUserGuide/run.md [hunk 4]
```diff
@@ -150,44 +150,47 @@ the dashboard URL is http://your_node:8888/
...
+* Yarn cluster mode (with conda package name "environment.tar.gz" for example)
 ```bash
 export SPARK_HOME=the root directory of Spark
 export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package
-export VENV_HOME=the parent directory of venv.zip and venv folder
+export ENV_HOME=the parent directory of your conda environment package
 
-PYSPARK_PYTHON=${VENV_HOME}/venv.zip/venv/bin/python ${ANALYTICS_ZOO_HOME}/bin/spark-submit-python-with-zoo.sh \
-    --conf spark.yarn.appMasterEnv.PYSPARK_PYTHON=venv.zip/venv/bin/python \
+PYSPARK_PYTHON=./environment/bin/python ${ANALYTICS_ZOO_HOME}/bin/spark-submit-python-with-zoo.sh \
```
Comment (hkvision) [inline docs/docs/PythonUserGuide/run.md]: Is environment.tar.gz uncompressed here?

Comment (jenniew) [reply docs/docs/PythonUserGuide/run.md]: No need to uncompress

Comment (hkvision) [reply docs/docs/PythonUserGuide/run.md]: Ar yes its worker python path, no need to uncompress

File: docs/docs/PythonUserGuide/run.md [hunk 5]
```diff
@@ -150,44 +150,47 @@ the dashboard URL is http://your_node:8888/
...
 
 Here we use Analytics Zoo [Object Detection Python example](https://github.com/intel-analytics/analytics-zoo/tree/master/pyzoo/zoo/examples/objectdetection) for illustration.
 
-* Yarn cluster mode
+* Yarn cluster mode (with conda package name "environment.tar.gz" for example)
 ```bash
 export SPARK_HOME=the root directory of Spark
 export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package
-export VENV_HOME=the parent directory of venv.zip and venv folder
+export ENV_HOME=the parent directory of your conda environment package
```
Comment (hkvision) [inline docs/docs/PythonUserGuide/run.md]: Then in cluster mode, ENV_HOME is unnecessary, the parent directory is never used. Only the path to the conda package is used.

**Comment by hkvision** [comment]
Fix #1989

File: docs/docs/PythonUserGuide/install.md (hunk 1 above)
Comment (jenniew) [reply docs/docs/PythonUserGuide/install.md]: Updated

File: docs/docs/PythonUserGuide/install.md (hunk 2 above)
Comment (jenniew) [reply docs/docs/PythonUserGuide/install.md]: Updated

File: docs/docs/PythonUserGuide/install.md (hunk 3 above)
Comment (jenniew) [reply docs/docs/PythonUserGuide/install.md]: Updated

File: docs/docs/PythonUserGuide/run.md (hunk 5 above)
Comment (jenniew) [reply docs/docs/PythonUserGuide/run.md]: Just use the ENV_HOME to specify the home directory

**Pull request merged**